examples/hitl/rearrange_v2/app_state_start_session.py

```diff
@@ -71,79 +79,171 @@ def _try_get_episode_ids(self) -> Optional[List[str]]:
         * Invalid 'episodes' format.
         * Episode indices out of bounds.
         """
-        data = self._app_data
 
         # Sanity checking.
-        if len(data.connected_users) == 0:
+        user_count = len(data.connected_users)
+        if user_count == 0:
             print("No user connected. Cancelling session.")
             return None
-        connection_record = list(data.connected_users.values())[0]
+        episodes: List[List[int]] = []
 
-        # Validate that episodes are selected.
-        if "episodes" not in connection_record:
-            print("Users did not request episodes. Cancelling session.")
-            return None
-        episodes_str = connection_record["episodes"]
+        # Validate that the episodes are integers.
+        for user_index in range(user_count):
+            episodes.append([])
+            connection_record = list(data.connected_users.values())[user_index]
+
+            # Validate that episodes are selected.
+            if "episodes" not in connection_record:
+                print("Users did not request episodes. Cancelling session.")
+                return None
+            episodes_str = connection_record["episodes"]
 
-        # Validate that all users are requesting the same episodes.
-        for connection_record in data.connected_users.values():
-            if connection_record["episodes"] != episodes_str:
+            # Validate that the parameter is a string.
+            if episodes_str is None or not isinstance(episodes_str, str):
                 print(
-                    "Users are requesting different episodes! Cancelling session."
+                    f"Episodes are supplied in an unexpected format: '{type(episodes_str)}'. Cancelling session."
                 )
                 return None
 
-        # Validate that the episode set is not empty.
-        if episodes_str is None or len(episodes_str) == 0:
-            print("Users did not request episodes. Cancelling session.")
-            return None
+            # Parse episode string.
+            episodes_split = episodes_str.split(",")
```

Comment:
Beware you are kinda duplicating the `episode_filter` logic in `hitl_driver.py`. Consider reusing that instead.

Reply:
This is a planned refactor post single-learn.

Lots of the code in the `rearrange_v2` can be moved to `habitat-hitl`.